### PR TITLE
SONOFF Zigbee device OTA support

### PR DESCRIPTION
**Pull request by lisongjun12**
1.add sonoff device OTA function.

### CODE REVIEW

**Comment by puddly** [comment]
Thank you for the pull request.

To ensure your code matches the zigpy project style (and tests pass), please install `pre-commit`:

```
pip install pre-commit
pre-commit install
pre-commit run --all-files  # to fix current issues
```

**Review by puddly — CHANGES_REQUESTED** [review]
Could you also a unit test including the JSON response sent by your OTA server?  If you modeled it after IKEA's, the unit tests should be almost identical.

File: zigpy/__init__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 MAJOR_VERSION = 0
-MINOR_VERSION = 51
```
[the remark this answers is not in the document]
Comment (lisongjun12) [reply zigpy/__init__.py:2]: OK, I will perform unit tests.
I have not modified the version number. After pulling the code, I see this value. What version number to revert to now?

File: zigpy/ota/provider.py [hunk 2]
```diff
@@ -410,6 +410,91 @@ async def refresh_firmware_list(self) -> None:
...
+
+    async def refresh_firmware_list(self) -> None:
+        if self._locks[LOCK_REFRESH].locked():
+            return
+
+        async with self._locks[LOCK_REFRESH]:
+            async with aiohttp.ClientSession(headers=self.HEADERS) as req:
+                url = self.config.get(CONF_OTA_SONOFF_URL, self.UPDATE_URL)
+                async with req.get(url) as rsp:
+                    # SONOFF does not always respond with an appropriate Content-Type
```
[the remark this answers is not in the document]
Comment (lisongjun12) [reply zigpy/ota/provider.py:473]: yes,

File: zigpy/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (puddly) [reply zigpy/__init__.py:2]: You can see in the diff (https://github.com/zigpy/zigpy/pull/1039/files) the previous value. I think you based your changes off of `master`, not `dev`.

[the remark this answers is not in the document]
Comment (lisongjun12) [reply zigpy/__init__.py:2]: ok, let me check.

**Comment by lisongjun12** [comment]
Switch to the dev branch to submit a PR, delete this branch.

**Pull request closed without merging**